src/dos/drives.cpp

```diff
@@ -311,6 +327,248 @@ char *DriveManager::GetDrivePosition(int drive)
 	return swap_position;
 }
 
+void DriveManager::SetDiskSpeed(DiskSpeed disk_speed, DiskType disk_type)
+{
+	switch (disk_type) {
+	case DiskType::Floppy: disk_settings.fdd_disk_speed = disk_speed; break;
+	case DiskType::HardDisk:
+		disk_settings.hdd_disk_speed = disk_speed;
+		break;
+	case DiskType::CdRom:
+		disk_settings.cdrom_disk_speed = disk_speed;
+		break;
+	default:
+		LOG_WARNING("DriveManager: Unknown disk type %d",
+		            static_cast<int>(disk_type));
+		return;
```

Comment:
We use the `default: assertm(false, "Invalid DiskType enum value");` for the "impossible" cases. Also remove the `return`.